write-fonts/src/graph.rs

```diff
@@ -232,11 +241,33 @@ impl Graph {
         self.assign_spaces_hb();
         self.sort_shortest_distance();
 
-        let n_overflows = self.find_overflows().len();
-        if n_overflows > 0 {
-            log::info!("packing failed with {n_overflows} overflows");
+        if !self.has_overflows() {
+            return true;
         }
-        n_overflows == 0
+
+        // now isolate spaces in a loop, until there are no more left:
+        let overflows = loop {
```

Comment:
The problem is that we aren't always going to continue until there are no overflows left; sometimes overflow is unavoidable.

The control flow here is:
- check if there are overflows
- if there are none, return
- if there are some, try to make progress on resolving them
- if we don't make progress, break the loop and print diagnostics

This should always terminate, since we exit the loop if either there are no overflows, *or* if there are overflows but there are no splittable subgraphs remaining.

Reply:
The comment above reads as if we are going to continue until there are none so that might need a fix then. IMHO while primary-exit-condition is still nicer, even if there is another path toward failure.

What about using `Result<(), SomeErrType>` so you can use normal control flow, e.g. `try_splitting_subgraphs(...)?`?